src/renderer/components/node/NodeInputs.tsx

```diff
@@ -31,53 +29,26 @@ interface FullInputProps extends Omit<Omit<Input, 'type'>, 'id'>, InputProps {
     definitionType: Type;
 }
 
-// TODO: perhaps make this an object instead of a switch statement
-const pickInput = (kind: InputKind, props: FullInputProps) => {
+const InputComponents: Readonly<
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    let InputType: React.MemoExoticComponent<(props: any) => JSX.Element> = GenericInput;
-    switch (kind) {
-        case 'file':
-            InputType = FileInput;
-            break;
-        case 'directory':
-            InputType = DirectoryInput;
-            break;
-        case 'text-line':
-            InputType = TextInput;
-            break;
-        case 'text':
-            InputType = TextAreaInput;
-            break;
-        case 'dropdown':
-            InputType = DropDownInput;
-            break;
-        case 'number':
-            InputType = NumberInput;
-            break;
-        case 'slider':
-            InputType = SliderInput;
-            break;
-        case 'generic':
-            return (
-                <InputContainer
-                    generic
-                    definitionType={props.definitionType}
-                    hasHandle={props.hasHandle}
-                    id={props.id}
-                    inputId={props.inputId}
-                    key={`${props.id}-${props.inputId}`}
-                    optional={props.optional}
-                >
-                    <GenericInput {...props} />
-                </InputContainer>
-            );
-        default:
-            return assertNever(kind);
-    }
+    Record<InputKind, React.MemoExoticComponent<(props: any) => JSX.Element>>
+> = {
+    file: FileInput,
+    directory: FileInput,
```

Comment:
Shit.

Reply:
I should probably review a bit more thoroughly lol